src/main/java/com/google/collegeplanner/servlets/PlannerServlet.java

```diff
@@ -29,35 +36,230 @@
 /** Servlet that returns the mutli-semester plan for the given courses.*/
 @WebServlet("/api/planner")
 public class PlannerServlet extends HttpServlet {
+  private HashMap<String, Integer> indegree; // Number of prerequisites for a course
+  private HashMap<String, HashSet<String>>
+      nextCourses; // Courses that come after the given course in the graph
+  private HashMap<String, HashSet<String>> corequisites;
+  private HashMap<String, Integer> heights; // height of a course in the graph
+  private HashMap<String, Integer> credits; // Number of credits for each course
+  private LinkedList<String> courseList;
+  private int totalCredits;
+
   /**
-   * Organizes courses into the given number of semesters
+   * Organizes courses from POST request into a given number of semesters
    */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // TODO(ramyabuva): Implement algorithm to split into the number of semesters
     JSONObject body;
+    int numSemesters;
+    JSONArray selectedClasses;
     try {
       body = getBody(request);
-    } catch (ParseException | NullPointerException e) {
+      numSemesters = Integer.parseInt((String) body.get("semesters"));
+      selectedClasses = (JSONArray) body.get("selectedClasses");
+    } catch (Throwable e) {
```

Comment:
generally, you should be catching concrete exceptions: https://screenshot.googleplex.com/RZ3ac61r01n